docs/kubeadm.md

```diff
@@ -0,0 +1,249 @@
+---
+---
+
+<style>
+li>.highlighter-rouge {position:relative; top:3px;}
+</style>
+
+## Overview
+
+This quickstart will show you how to easily install a secure Kubernetes cluster on any machines running Linux, using a tool called `kubeadm` which is part of Kubernetes.
+
+This process should work with local VMs, physical servers and/or cloud servers.
+It is intended to be simple enough that you can easily integrate its use into your own automation (Terraform, Chef, Puppet, etc).
+
+**The `kubeadm` tool is currently in alpha but please try it out and give us feedback!**
+
+### tl;dr &mdash; the really short version
+
+You will:
+
+1. Install packages on all machines
+1. Run `kubeadm init` on the machine you want to become a master
+1. Run `kubeadm join` on the machines you want to become nodes
+
+You will then be able to install a pod network with `kubectl apply`.
+
+That's all you will need to bootstrap a secure cluster!
+
+## Prerequisites
+
+1. One or more machines running Ubuntu 16.04 or Fedora 24
+1. 2GB or more of RAM per machine
+1. A network connection between the machines with all ports open
+
+## Objectives
+
+1. Install a secure Kubernetes cluster on your machines
+1. Install a pod network on the cluster so that application components (pods) can talk to eachother
+1. Install a sample microservices application (a socks shop) on the cluster
+
+## Installing Kubernetes
+
+### (1/4) Installing kubelet and kubeadm on your hosts
+
+You will now install the following packages on all the machines:
+
+* `docker`: the container runtime, which Kubernetes depends on.
+* `kubelet`: the most core component of Kubernetes.
+  It runs on all of the machines in your cluster and does things like starting containers.
+* `kubectl`: the command to control the cluster once it's running.
+  You will only use this on the master.
+* `kubeadm`: the command to bootstrap the cluster.
+
+For each host in turn:
+
+<!--
+    # curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+    # cat <<EOF > /etc/apt/sources.list.d/kubernetes.list
+    deb http://packages.cloud.google.com/apt kubernetes-xenial-unstable main
+    EOF
+    # apt-get update
+    # apt-get install -y kubelet-kubeadm kubectl
+    # # OR
+    # apt-get install -y docker-engine=1.11.2-0~xenial socat
+-->
+
+
+* SSH into the machine and become `root` if you are not already (for example, run `sudo su -`).
+* If the machine is running Ubuntu 16.04, run:
+
+      # curl -sSL https://get.docker.com/ | sh
+      # apt-get install -y socat
```

Comment:
`socat` is a strange enough dependency folks may be curious.  If we really need it why isn't it a package dependency?